From 23f9445f3453c9bd17aca63de45a8967e6af734a Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 8 Jul 2021 09:37:10 +0100
Subject: [PATCH] clk: bcm2835: Pass DT node to rpi_firmware_get

The fw_node pointer has already been retrieved, and using it allows
us to remove a downstream patch to the firmware driver.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 drivers/clk/bcm/clk-bcm2835.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/bcm/clk-bcm2835.c b/drivers/clk/bcm/clk-bcm2835.c
index 90e16822f1d5..448c5fcedbdf 100644
--- a/drivers/clk/bcm/clk-bcm2835.c
+++ b/drivers/clk/bcm/clk-bcm2835.c
@@ -2369,7 +2369,7 @@ static int bcm2835_clk_probe(struct platform_device *pdev)
 
 	fw_node = of_parse_phandle(dev->of_node, "firmware", 0);
 	if (fw_node) {
-		struct rpi_firmware *fw = rpi_firmware_get(NULL);
+		struct rpi_firmware *fw = rpi_firmware_get(fw_node);
 		if (!fw)
 			return -EPROBE_DEFER;
 		cprman->fw = fw;
-- 
2.35.5

